drivers/pwm/pwm_nrfx.c

```diff
@@ -286,8 +286,13 @@ static int pwm_nrfx_init(const struct device *dev)
 static void pwm_nrfx_uninit(const struct device *dev)
 {
 	const struct pwm_nrfx_config *config = dev->config;
+	struct pwm_nrfx_data *data = dev->data;
 
 	nrfx_pwm_uninit(&config->pwm);
+
+	data->period_cycles = 0;
+	data->countertop = 0;
+	data->prescaler = 0;
```

Comment:
`data->current[]` should also be zeroed, otherwise `pwm_period_check_and_set()` will fail on this check:
https://github.com/zephyrproject-rtos/zephyr/blob/4ffde634a8ceece1eeb92d8164ccf11c61659577/drivers/pwm/pwm_nrfx.c#L50-L56
if multiple channels are used and they are not turned off before suspending the PWM.

Reply:
Fixed - it becomes one-liner now.